binaries/cuprated/src/config.rs

```diff
@@ -76,32 +93,69 @@ pub fn read_config_and_args() -> Config {
     args.apply_args(config)
 }
 
-/// The config for all of Cuprate.
-#[derive(Debug, Default, Deserialize, Serialize, PartialEq)]
-#[serde(deny_unknown_fields, default)]
-pub struct Config {
-    /// The network we should run on.
-    network: Network,
-
-    pub no_fast_sync: bool,
-
-    /// [`tracing`] config.
-    pub tracing: TracingConfig,
-
-    pub tokio: TokioConfig,
-
-    pub rayon: RayonConfig,
-
-    /// The P2P network config.
-    p2p: P2PConfig,
-
-    /// The storage config.
-    pub storage: StorageConfig,
+config_struct! {
+    /// The config for all of Cuprate.
+    #[derive(Debug, Deserialize, Serialize, PartialEq)]
+    #[serde(deny_unknown_fields, default)]
+    pub struct Config {
+        /// The network we should run on.
+        ///
+        /// Valid values: "Mainnet", "Testnet" and "Stagenet".
```

Comment:
Formatted like this elsewhere.
```suggestion
        /// Valid values: ["Mainnet", "Testnet" and "Stagenet"]
```